docs/administration/datagrid/fields.md

```diff
@@ -0,0 +1,83 @@
+# Fields
+
+Fields are the building blocks of a datagrid. They are used to define the columns of the datagrid and to specify how the data should be displayed.
+
+## Configuration
+
+Fields are defined by the `Datagrid` class using the `add(string $name, array $options)`, `update(string $name, array $options)` and `remove(string $name)` methods. The `$name` parameter is the name of the field and the `$options` parameter is an array of options that define the behavior of the field.
+
+In case you are using `OrmAdapter`, you can use dot notation to access nested properties of the entity. Here are some examples of how to define field names:
+
+- `$datagrid->add('domainId')` - This will fetch the `domainId` property of the entity.
+- `$datagrid->add('currency.code')` - This will join the `currency` relation and fetch the `code` property of the `Currency` entity.
+- `$datagrid->add('status.name')` - This will join the `status` relation and look for the `name` property of the `Status` entity. In this case `Status` is a translatable entity and the `name` property is a translatable field. The datagrid will automatically fetch the correct translation based on the current locale.
+- `$datagrid->add('currency')` - This will fetch the `Currency` entity. In this case you need to define `template`, `transform` or `visible` to `false` option to tell datagrid how to display this object.
+
+### Options
+
+The following options are available for fields:
+
+- `label` - The label of the field. It is displayed in the header of the column.
+- `visible` - If set to `true`, the field is displayed in the datagrid. If set to `false`, the field is hidden, but fetching the data is still performed.
+- `sortable` - If set to `true`, the field can be sorted by clicking on the column header.
+- `virtual` - If set to `true`, the field is not fetched from the data source.
+- `help` - The help text that is displayed next to label in column header.
+- `template` - The template that is used to render the field. The template is a path to a Twig template file.
+- `transform` - A callback function that is used to transform the data before it is displayed in the field. The callback function receives the value of the field as the first parameter, row as the second argument and all rows as the third parameter.
```

Comment:
If I use transform on a virtual field I get fatal error

code:
```
 protected function configureDatagrid(Datagrid $datagrid): Datagrid
    {
        $datagrid->add('name');

        $datagrid->add('oreo', [
            'label' => 'Oreo',
            'virtual' => true,
            'transform' => function () {
                return 'asd';
            },
        ]);

        $datagrid->setDefaultOrder('name', OrderingEnum::DESC);
        $datagrid->setPagination(false);

        return $datagrid;
    }
```

![Snímek obrazovky 2025-03-05 v 10 15 30](https://github.com/user-attachments/assets/d7785732-3b3b-44d3-b83e-d772bd502e80)

Reply:
Maybe we can discuss how this should work. Like You want to transform data that does not exists :D

Reply:
But I agree that there should be some check that will tell you something like `You cannot transform virtual field`

Reply:
Ok, I made it fully functional. In case you want to transforma  virtual field, in closure you will get null instead, but you can return what you want to show based on other fields, etc.